addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/model/Realtime.java

```diff
@@ -18,31 +16,51 @@
  */
 public class Realtime extends ErrorResponse {
 
-    @SerializedName(value = "current", alternate = "current_ma")
+    private static final int V2_TO_V1 = 1000;
+
     private double current;
-    @SerializedName(value = "power", alternate = "power_mw")
     private double power;
-    @SerializedName(value = "total", alternate = "total_wh")
     private double total;
-    @SerializedName(value = "voltage", alternate = "voltage_mv")
     private double voltage;
 
+    // JSON names used for v2 hardware
+    private double current_ma;
+    private double power_mw;
+    private double total_wh;
+    private double voltage_mv;
+
     public double getCurrent() {
+        if (current == 0 && current_ma != 0) {
+         return current_ma / V2_TO_V1; // v1 displays as amps
+        }
+
         return current;
     }
 
     public double getPower() {
+        if (power == 0.0 && power_mw != 0.0) {
+          return power_mw / V2_TO_V1; // v1 displays as watts
+        }
+
         return power;
     }
 
     public double getTotal() {
+        if (total == 0.0 && total_wh != 0.0) {
+          return total_wh / V2_TO_V1; // v1 displays as kWh
+        }
+
         return total;
     }
 
     public double getVoltage() {
+        if (total == 0.0 && voltage_mv != 0.0) {
+          return voltage_mv / V2_TO_V1; // v1 displays as volts
+        }
         return voltage;
     }
 
+
     @Override
     public String toString() {
         return "current:" + current + ", power:" + power + ", total:" + total + ", voltage:" + voltage
```

Comment:
Please update the toString: just replace values with the methods: `current` -> `getCurrent`.